.github/workflows/build.yml

```diff
@@ -35,11 +35,15 @@ jobs:
           path: ~/.sonar/cache
           key: ${{ runner.os }}-sonar
           restore-keys: ${{ runner.os }}-sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        if: ${{ env.SONAR_TOKEN != '' }}
       - name: Maven Build and Analyze
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         run: mvn clean install -B sonar:sonar -Dsonar.projectKey=SAP_xsk
+        if: ${{ env.SONAR_TOKEN != '' }}
```

Comment:
Please put the `if` expression on line `42` just before `env:`, that way it would be more readable.